quiz-bundle-spec.md

```diff
@@ -0,0 +1,71 @@
+# Qwiklabs Quiz Bundle Specification
+
+**Version 1**
+
+> This is a DRAFT document. We welcome feedback as this format evolves.
+
+## `qwiklabs.yaml` Structure
+
+Here's a sample `qwiklabs.yaml` file with all nested details removed to make it easier to see the general file structure.
+
+```yml
+entity_type: Quiz
+schema_version: 1
+
+# The authoring title of the quiz - not visible to learners.
+title: Super Hard Quiz
+
+# The threshold grade that a student needs to achieve in order to count as "passing" the quiz.
+passing_percentage: 67
+
+# The "items" (i.e. questions) contained in this quiz.
+items: ...
+```
+
+Note that all of the localized content (stem, option titles, rationales, etc) are HTML content that may be displayed in various contexts. All of these chunks will be sanitized according to the restricted set in the [HTML spec](./html-spec.md).
+
+### Items
+Items are polymorphic - i.e. there are several different _item types_ that are defined slightly differently. `items` is an array of items defined correctly for their type as defined by the `type` attribute.
+
+#### Multiple Choice Items
+
+A quiz item that has multiple options and one answer. One option is the _answer_, and the rest are _distractors_. Each option has a required rationale that explains why that option is correct or incorrect.
+
+attribute               | required | type              | notes
+----------------------- | -------- | ----------------- | -----------------------------------------
+id                      | ✓        | string            | A unique identifier for this item
+type                    | ✓        | 'multiple-choice' | The item type, which is always 'multiple-choice'
+stem                    | ✓        | dictionary        | A locale dictionary of locales for the text that asks the question, such as "Which of the following is a color?"
+options                 | ✓        | array             | An array of item options
```

Comment:
Consider moving the `id`, `title`, `rationale`, `is_answer` to an "option spec" section below and removing it from here and the `multiple-select`

Reply:
I thought about that - I thought it might be a bit confusing to authors. Then again, this is more machine generated / readable. I'll switch to that.

Reply:
I'm all for this suggestion. You should feel free to assume that the readers of this document are software engineers or at least Lab Architects. Content authors will read the corresponding GitWhisperer (or other integrating system's) documentation.

Along those lines feel free to get really specific about type definition...

```suggestion
options                 | ✓        | Array of Options  | Unordered list of options for the question. See [mulitple choice options spec](#multiple-choice-options).
```